pghoard/rohmu/object_storage/sftp.py

```diff
@@ -0,0 +1,209 @@
+"""
+rohmu - sftp object store interface
+
+Copyright (c) 2016 Ohmu Ltd
+See LICENSE for details
+"""
+
+import logging
+import warnings
+import pysftp
+from pysftp import CnOpts
+import os
+import datetime
+from stat import S_ISDIR
+from io import BytesIO, StringIO
+import json
+from .base import BaseTransfer, IterKeyItem, KEY_TYPE_PREFIX, KEY_TYPE_OBJECT
+from ..errors import FileNotFoundFromStorageError, InvalidConfigurationError, StorageError
+
+
+class SFTPTransfer(BaseTransfer):
+    def __init__(self,
+                 server,
+                 port,
+                 username,
+                 password=None,
+                 private_key=None,
+                 prefix=None):
+        super().__init__(prefix=prefix)
+        self.server = server
+        self.port = port
+        self.username = username
+        self.password = password
+        self.private_key = private_key
+
+        if password is None and private_key is None:
```

Comment:
`if not password and not private_key:`? I'm guessing empty string isn't better than None.

Reply:
This would mean that the sftp server supports empty passwords, which I guess is possible, I will check.